src/app/command-channel.service.ts

```diff
@@ -68,9 +68,18 @@ export class CommandChannelService implements OnDestroy {
   connect(clientUrl: string): Observable<void> {
     const ret = new Subject<void>();
     this.api.getToken()
-      .pipe(first())
-      .subscribe(token => {
-        this.ws = new WebSocket(clientUrl, `base64url.bearer.authorization.containerjfr.${btoa(token)}`);
+      .pipe(
+        combineLatest(this.api.getAuthMethod()),
+        first()
+      )
+      .subscribe(auths => {
+        let subprotocol = undefined;
```

Comment:
This feels a bit weird... but it makes sense. Just to point out for the 2nd arg of `WebSocket` constructor, `null`, `undefined` and `""` are equivalent.

Reply:
Yea I felt a bit weird about it too, but using the empty string or null seemed like it might look like a mistake on later review. `undefined` was weird enough to look intentional, I guess.